src/firefly/js/ui/UploadTableSelector.jsx

```diff
@@ -0,0 +1,460 @@
+ */
+
+const columnFieldsType = PropTypes.arrayOf(PropTypes.shape({
+    ...omit({...ColumnFld.propTypes}, ['cols']), //because cols come from uploadInfo
+    guessValue: PropTypes.func //(columns) => string
+}));
+
+UploadTableSelector.propTypes = {
+    uploadInfo: PropTypes.object,
+    setUploadInfo: PropTypes.func,
+    columnFields: columnFieldsType,
+    columnMappingPanelKey: PropTypes.string,
+    allowUploadColumnsSelection: PropTypes.bool,
+    defaultUploadColumnsSelection: PropTypes.shape({
+        colTypes: PropTypes.arrayOf(PropTypes.string),
+        colCount: PropTypes.number
+    }),
+    slotProps: PropTypes.shape({
+        fileInfo: PropTypes.object,
+        columnMappingPanel: PropTypes.shape({
+            component: PropTypes.elementType,
+            ...ColumnMappingPanel.propTypes
+        })
+    })
+};
+
+/**
+ * A generic component that shows a collapsible panel for mapping columns, with a header and column input fields.
+ * Note: this a controlled component whose state is managed by the parent component, generally an UploadTableSelector.
+ *
+ * @param props
+ * @param props.cols {Columns[]} - All available columns to select mappings from
+ * @param props.columnFieldValues {string[]} - array of field values, one for each column field in FieldGroupState
+ * @param props.columnFields {ColumnFieldsType[]} - array of objects that define the column fields: fieldKey, name are
+ * required keys, guessValue and any key that can be passed to ColumnFld as props is optional
+ * @param props.panelKey {string} - key for the collapsible panel that contains column mapping
+ * @param props.headerTitle {Node} - (prefix) title of the header
+ * @param props.getHeaderColumnMapping {function(string[]):string} - function to render the text in header for column mapping
+ * @param props.headerPostTitle {string} - (suffix) title text to show after the column mapping in header
+ * @param props.openPreMessage {string} - message to show before column input fields when the panel is open
+ * @param props.sx {Object} - sx prop used by the FieldGroupCollapsible
+ * @param props.slotProps {Object} - slotProps for the component
+ * @param props.children {Node} - children to render inside the panel. Note: this will override the default rendering
+ * of column input fields
+ * @returns {Element}
+ */
+export function ColumnMappingPanel({cols, columnFieldValues, columnFields, panelKey,
+                                       headerTitle='Mapped Columns:', getHeaderColumnMapping, headerPostTitle = '',
+                                       openPreMessage='', sx, slotProps, children}) {
+    const defaultColumnMapping = (colValues) => colValues?.every((val) => !val)
+        ? 'unset'
+        : colValues?.map((val) => val || 'unset').join(', ');
+
+    const panelHeader= (
+        <Stack {...{direction:'row', alignItems:'baseline', spacing:1}}>
+            <span>{headerTitle}</span>
```

Comment:
Interesting how you choose to use `span` only for this one when you are using Typography everywhere else.

Reply:
This was a copy-over from older deleted files. My interpretation of this choice was: headerTitle can be a `node` (with inline display type) that's why `span` is used, whereas following header bits are `string`. 

Can `Typography` be used to wrap a node type component?  if yes, I'll update it

Reply:
I thought the same, until I noticed that headerTitle is declared as a String.
```
headerTitle: PropTypes.string,
```

Reply:
Ah, good catch - I'll fix the prop type